### PR TITLE
*: support signature push down.

### PR DESCRIPTION
**Pull request by hanfei1991**
1. push more signatures, open the switch.
2. like always accepts 3 args.
3. add type info to column and constant pb.

### CODE REVIEW

**Comment by hanfei1991** [comment]
@XuHuaiyu @zz-jason PTAL

File: expression/builtin.go [hunk 1]
```diff
@@ -916,7 +916,7 @@ var funcs = map[string]functionClass{
 	ast.UnaryMinus: &unaryMinusFunctionClass{baseFunctionClass{ast.UnaryMinus, 1, 1}},
 	ast.IsTruth:    &isTrueOrFalseFunctionClass{baseFunctionClass{ast.IsTruth, 1, 1}, opcode.IsTruth},
 	ast.IsFalsity:  &isTrueOrFalseFunctionClass{baseFunctionClass{ast.IsFalsity, 1, 1}, opcode.IsFalsity},
-	ast.Like:       &likeFunctionClass{baseFunctionClass{ast.Like, 2, 3}},
+	ast.Like:       &likeFunctionClass{baseFunctionClass{ast.Like, 3, 3}},
```
Comment (zz-jason) [inline expression/builtin.go]: bug fix ?

**Comment by XuHuaiyu** [comment]
please resolve the conflicts.
rest LGTM

File: expression/builtin.go (hunk 1 above)
Comment (hanfei1991) [reply expression/builtin.go]: no, it's a compatible problem.

File: store/tikv/coprocessor.go [hunk 2]
```diff
@@ -83,7 +83,8 @@ func (c *CopClient) supportExpr(exprType tipb.ExprType) bool {
 	case kv.ReqSubTypeDesc:
 		return true
 	case kv.ReqSubTypeSignature:
-		return c.store.mock
+		//return c.store.mock
```
Comment (shenli) [inline store/tikv/coprocessor.go]: Remove this.

**Comment by winoros** [comment]
@coocood Currently `tikv=` can only use master/pre-ga/rc4... branch.

**Comment by shenli** [comment]
@hanfei1991 Please resolve the conflicts and remove the DNM label.

**Pull request merged**